cmd/sealer/cmd/cluster/run.go

```diff
@@ -73,6 +77,43 @@ func NewRunCmd() *cobra.Command {
 				clusterFile     = runFlags.ClusterFile
 				applyMode       = runFlags.Mode
 			)
+			imageEngine, err := imageengine.NewImageEngine(imagecommon.EngineGlobalConfigurations{})
+			if err != nil {
+				return err
+			}
+
+			extension, err := imageEngine.GetSealerImageExtension(&imagecommon.GetImageAnnoOptions{ImageNameOrID: args[0]})
+			if err != nil {
+				return fmt.Errorf("failed to get cluster image extension: %s", err)
+			}
+
+			if extension.Type == v12.AppInstaller {
+				logrus.Infof("start to install application: %s", args[0])
+				cf, err := clusterfile.NewClusterFile(nil)
+				if err != nil {
+					return err
+				}
+
+				cluster := cf.GetCluster()
+				infraDriver, err := infradriver.NewInfraDriver(&cluster)
+				if err != nil {
+					return err
+				}
+
+				if err := imageEngine.Pull(&imagecommon.PullOptions{
+					Quiet:      false,
+					PullPolicy: "missing",
+					Image:      args[0],
+					Platform:   "local",
+				}); err != nil {
+					return err
+				}
+				return installApplication(args[0], nil, extension, infraDriver, imageEngine, applyMode)
```

Comment:
not need to pass nil if you have added the error message prompt for non-app image with `LaunchCmds`